easybuild/scripts/rpath_args.py

```diff
@@ -0,0 +1,119 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Utility script used by RPATH wrapper script;
+output is statements that define the following environment variables
+* $CMD_ARGS: new list of command line arguments to pass
+* $RPATH_ARGS: command line option to specify list of paths to RPATH
+
+author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+import re
+import sys
+
+
+cmd = sys.argv[1]
+rpath_filter = sys.argv[2]
+args = sys.argv[3:]
+
+# wheter or not to use -Wl to pass options to the linker
+if cmd in ['ld', 'ld.gold']:
+    flag_prefix = ''
+else:
+    flag_prefix = '-Wl,'
+
+if rpath_filter:
+    rpath_filter = rpath_filter.split(',')
+else:
+    # don't RPATH system library locations by default
+    rpath_filter = ['/lib', '/usr/lib']
+
+version_mode = False
+cmd_args, cmd_args_rpath = [], []
+
+# process list of original command line arguments
+idx = 0
+while idx < len(args):
+
+    arg = args[idx]
+
+    # if command is run in 'version check' mode, make sure we don't include *any* -rpath arguments
+    if arg in ['-v', '-V', '--version', '-dumpversion']:
+        version_mode = True
+        cmd_args.append(arg)
+
+    # FIXME: also consider $LIBRARY_PATH?
+    # FIXME: support to hard inject additional library paths?
+    # FIXME: support to specify list of path prefixes that should not be RPATH'ed into account?
+    # FIXME skip paths in /tmp, build dir, etc.?
+
+    # handle -L flags, inject corresponding -rpath flag
+    elif arg.startswith('-L'):
+        # take into account that argument to -L may be separated with one or more spaces...
+        if arg == '-L':
+            # actual library path is next argument when arg='-L'
+            idx += 1
+            lib_path = args[idx]
+        else:
+            lib_path = arg[2:]
+
+        if os.path.isabs(lib_path) and not any(lib_path.startswith(f) for f in rpath_filter):
+            # inject -rpath flag in front for every -L with an absolute path,
+            # also retain the -L flag (without reordering!)
+            cmd_args_rpath.append(flag_prefix + '-rpath=%s' % lib_path)
+            cmd_args.append('-L%s' % lib_path)
+        else:
+            # don't RPATH in relative paths;
+            # it doesn't make much sense, and it can also break the build because it may result in reordering lib paths
+            cmd_args.append('-L%s' % lib_path)
+
+    # filter out --enable-new-dtags if it's used;
+    # this would result in copying rpath to runpath, meaning that $LD_LIBRARY_PATH is taken into account again
+    elif arg != '--enable-new-dtags':
+        cmd_args.append(arg)
+
+    idx += 1
+
+# add -rpath flags in front
+cmd_args = cmd_args_rpath + cmd_args
+
+if not version_mode:
```

Comment:
see above, commands like `gcc -v` should not get broken by injecting `-rpath` flags